test/Google.Api.Gax.Tests/PlatformTest.cs

```diff
@@ -104,5 +104,65 @@ public void Gke_WithData(bool withPodName)
             Assert.Equal("de3bf4eb-b036-11e7-8e33-42010a9a0fdd", gke.PodId);
             Assert.Equal("platformintegrationtest", gke.ContainerName);
         }
+
+        const string metadataValid = "{'project':{'projectId':'FakeProjectId'},'instance':{'attributes':{'cluster-name':'FakeClusterName'},'id':'123','zone':'projects/FakeProject/zones/FakeLocation'}}";
+        const string metadataIncomplete = "{'project':{'projectId':'FakeProjectId'},'instance':{'attr";
+        const string namespaceValid = "{'kind':'Namespace','metadata':{'uid':'namespaceid'}}";
+        const string namespaceWrongKind = "{'kind':'NotNamespace','metadata':{'uid':'namespaceid'}}";
+        const string namespaceIncomplete = "{'kind':'Namespace','m";
+        const string podValid = "{'kind':'Pod','metadata':{'name':'podname','uid':'podid'}}";
+        const string podWrongKind = "{'kind':'NotPod','metadata':{'name':'podname','uid':'podid'}}";
+        const string podIncomplete = "{'kind':'Pod','metadata':{";
```

Comment:
```csharp
const string podNoKind = "{'unkind': 'Pod'}";
```
(and ditto for `namespace`) to test my earlier objection.

Reply:
Done